src/tests/test_rest_framework_dso/test_serializers.py

```diff
@@ -69,21 +74,25 @@ def test_serializer_embed_with_missing_relations(drf_request):
     cursor.execute(
         "INSERT INTO test_rest_framework_dso_movie (name, category_id) VALUES ('Test', 333);"
     )
-    movie = Movie.objects.get(name="Test")
-
-    serializer = MovieSerializer(
-        many=True,
-        instance=[movie],
-        fields_to_expand=["category"],
-        context={"request": drf_request},
-    )
-    assert serializer.data == {
-        "movie": [{"name": "Test", "category_id": movie.category_id}],
-        "category": [],
-    }
-
-    # Cleanup needed to make Django happy.
-    cursor.execute("DELETE FROM test_rest_framework_dso_movie")
+    try:
+        movie = Movie.objects.get(name="Test")
+
+        serializer = MovieSerializer(
+            Movie.objects.all(),
+            many=True,
+            fields_to_expand=["category"],
+            context={"request": drf_request},
+        )
+
+        data = normalize_data(serializer.data)
+        assert data == {
+            "movie": [{"name": "Test", "category_id": movie.category_id}],
+            "category": [],
+        }
+    finally:
+        # Make sure the cleanup happens or unit tests report this error
+        # instead of the actual assertion error.
+        cursor.execute("DELETE FROM test_rest_framework_dso_movie WHERE category_id=333")
```

Comment:
Why is this cleanup necessary? I would have expected `@pytest.mark.django_db` to rollback the transaction at the end of this test function automatically?

Reply:
This code already existed before, only without a try..finally. At the top of the test, an invalid `category_id` is inserted. When resetting the database between tests, this causes a FK constraint error - which distracts from the actual test failure.